examples/jax/encoder/test_multigpu_encoder.py

```diff
@@ -0,0 +1,398 @@
+    dataset_size = len(dataset['sentence'])
+    output = np.zeros((dataset_size, max_seq_len), dtype=np.int32)
+
+    for j, sentence in enumerate(dataset['sentence']):
+        tokens = nltk.word_tokenize(sentence.decode("utf-8"))
+        tensor = output[j]
+
+        for i, word in enumerate(tokens):
+            if i >= max_seq_len:
+                break
+
+            if word not in vocab:
+                vocab[word] = word_id
+                tensor[i] = word_id
+                word_id = word_id + 1
+            else:
+                tensor[i] = vocab[word]
+
+    dataset['sentence'] = output
+    dataset['label'] = dataset['label'].astype(np.float32)
+    return dataset, vocab, word_id
+
+
+def get_datasets(max_seq_len):
+    """Load GLUE train and test datasets into memory."""
+    vocab = {}
+    word_id = 0
+    dataset = 'glue/cola'
+    train_ds = tfds.as_numpy(tfds.load(dataset, split='train', batch_size=-1))
+    train_ds, vocab, word_id = data_preprocess(train_ds, vocab, word_id, max_seq_len)
+    test_ds = tfds.as_numpy(tfds.load(dataset, split='validation', batch_size=-1))
+    test_ds, vocab, word_id = data_preprocess(test_ds, vocab, word_id, max_seq_len)
+    return train_ds, test_ds, word_id
+
+
+def check_fp8(state, var_collect, input_shape, label_shape):
+    "Check if model includes FP8."
+    inputs = jnp.zeros(input_shape, dtype=jnp.int32)
+    labels = jnp.empty(label_shape, dtype=jnp.bfloat16)
+    rngs = {DROPOUT_KEY: jax.random.PRNGKey(0)}
+    assert "Float8" in str(
+        jax.make_jaxpr(train_step, static_argnums=5)(state, inputs, labels, var_collect, rngs,
+                                                     True))
+
+
+def get_params_pspec(sharding_rules, abs_var_collect):
+    """Refer params to create params partition spec"""
+    rules_dict = {}
+    for key, value in sharding_rules:
+        rules_dict[key] = value
+
+    def to_device_axis(logical_axis):
+        partitions = [rules_dict[key] for key in logical_axis]
+        return jax.sharding.PartitionSpec(*partitions)
+
+    params_axes = abs_var_collect.get(PARAMS_AXES_KEY, {})
+    params_axes_pspec = jax.tree_map(to_device_axis, nn.partitioning.get_axis_names(params_axes))
+    params_pspec = jax.tree_map(lambda x: jax.sharding.PartitionSpec(), abs_var_collect[PARAMS_KEY])
+    params_pspec = FrozenDict({**params_pspec, **params_axes_pspec})
+    return params_pspec
+
+
+def get_state_pspec(state, params_pspec):
+    """Refer params_pspec to create state partition spec"""
+
+    def replace_params(x):
+        return params_pspec if isinstance(x, FrozenDict) else None
+
+    state_pspec = jax.tree_map(replace_params, state, is_leaf=lambda x: isinstance(x, FrozenDict))
+    return state_pspec
+
+
+def train_and_evaluate(args):
+    """Execute model training and evaluation loop."""
+    print(args)
+    assert num_gpu_is_enough(), "Need 8 GPUs to run"
+    assert args.batch_size % NUM_GPU == 0, "Batch size needs to be multiple of 8"
```

Comment:
Replace `"Batch size needs to be multiple of 8"`
with `"Batch size needs to be multiple of " + NUM_GPU`
or `"Batch size needs to be multiple of the number of GPUs"`

Reply:
done